### PR TITLE
fix:Fit `Binominal calculate.cpp` to guidelines

### PR DESCRIPTION
**Pull request by ewd00010**
#### Description of Change
FIX: altered the structure of if else if statements to have guard clauses first before main functionality, added brackets to make calculation steps more clear (Following BODMAS), updated namespaces used in file, removed unneeded user input code

CHORE: Added explanations and other documentation

#### Checklist
- [x] Added description of change
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: still no clangd working for me, file may need checked 
I'm pretty sure I could do a recursive version of this code, should I make one and add it as a function to this file?

### CODE REVIEW

**Comment by ewd00010** [comment]
maybe we should change the file name to just be `binominal_coefficient.cpp` aswell

File: math/binomial_calculate.cpp [hunk 1]
```diff
@@ -3,90 +3,104 @@
  * @brief Program to calculate [Binomial
  * coefficients](https://en.wikipedia.org/wiki/Binomial_coefficient)
  *
+ * @details
+ * quoted from: [Weisstein, Eric W. "Binomial Coefficient." From MathWorld--A
+ * Wolfram Web
+ * Resource.](https://mathworld.wolfram.com/BinomialCoefficient.html) "The
+ * binomial coefficient [; meaning n above k in brackets, how binomial
+ * coefficients are notated] (n; k) is the number of ways of picking k unordered
```
Comment (realstealthninja) [inline math/binomial_calculate.cpp:11]:
```suggestion
 * binomial coefficient \f$ \binom{n}{k} \f$ is the number of ways of picking \f$k\f$ unordered
```

File: math/binomial_calculate.cpp [hunk 2]
```diff
@@ -3,90 +3,104 @@
...
+ * outcomes from n possibilities, also known as a combination or combinatorial
```
Comment (realstealthninja) [inline math/binomial_calculate.cpp:12]:
```suggestion
 * outcomes from \f$n\f$ possibilities, also known as a combination or combinatorial
```

File: math/binomial_calculate.cpp [hunk 3]
```diff
@@ -3,90 +3,104 @@
...
+ * number. The symbols [ _ representing subscript ] _nC_k and (n; k) are used to
```
Comment (realstealthninja) [inline math/binomial_calculate.cpp:13]:
```suggestion
 * number. The symbols \f$ _nC_k \f$ and \f$ \binom{n}{k} \f$ are used to
```

File: math/binomial_calculate.cpp [hunk 4]
```diff
@@ -3,90 +3,104 @@
...
+ * denote a binomial coefficient, and are sometimes read as "n choose k." (n; k)
```
Comment (realstealthninja) [inline math/binomial_calculate.cpp:14]:
```suggestion
 * denote a binomial coefficient, and are sometimes read as "n choose k." \f$ \binom{n}{k} \f$
```

File: math/binomial_calculate.cpp [hunk 5]
```diff
@@ -3,90 +3,104 @@
...
+ * therefore gives the number of k-subsets possible out of a set of n distinct
+ * items. For example, The 2-subsets of {1,2,3,4} are the six pairs {1,2},
+ * {1,3}, {1,4}, {2,3}, {2,4}, and {3,4}, so (4; 2)=6." [Another good
+ * example/explanation](https://math.stackexchange.com/questions/2172355/probability-notation-two-numbers-stacked-inside-brackets)
+ *
+ * @note An identity of the binomial coefficient is (n; n-k) This is explained
```
Comment (realstealthninja) [inline math/binomial_calculate.cpp:20]:
```suggestion
 * @note An identity of the binomial coefficient is \f$ \binom{n}{n-k} \f$ This is explained
```

File: math/binomial_calculate.cpp [hunk 6]
```diff
@@ -3,90 +3,104 @@
...
  * @brief Mathematical algorithms
  */
 namespace math {
-/**
- * @namespace binomial
- * @brief Functions for [Binomial
- * coefficients](https://en.wikipedia.org/wiki/Binomial_coefficient)
- * implementation
- */
-namespace binomial {
```
[the remark this answers is not in the document]
Comment (ewd00010) [reply math/binomial_calculate.cpp]: reverted

File: math/binomial_calculate.cpp [hunk 7]
```diff
@@ -3,19 +3,31 @@
...
 #include <cassert>   /// for assert
-#include <cstdint>   /// for int32_t type
-#include <cstdlib>   /// for atoi
 #include <iostream>  /// for IO operations
 
-/**
- * @namespace math
- * @brief Mathematical algorithms
- */
-namespace math {
```
Comment (Panquesito7) [inline math/binomial_calculate.cpp:18]: Please add the `math` namespace back.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by alexpantyukhin — APPROVED** [review]

File: math/binomial_calculate.cpp [hunk 8]
```diff
@@ -3,44 +3,73 @@
...
 /**
  * @namespace binomial
  * @brief Functions for [Binomial
  * coefficients](https://en.wikipedia.org/wiki/Binomial_coefficient)
  * implementation
  */
-namespace binomial {
+namespace math { namespace binomial {
```
Comment (Panquesito7) [inline math/binomial_calculate.cpp:37]: Please add the `math` namespace documentation back.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: math/binomial_calculate.cpp [hunk 9]
```diff
@@ -3,44 +3,76 @@
...
-namespace math {
-/**
+ *
  * @namespace binomial
  * @brief Functions for [Binomial
  * coefficients](https://en.wikipedia.org/wiki/Binomial_coefficient)
  * implementation
  */
-namespace binomial {
+namespace math { namespace binomial {
```
Comment (realstealthninja) [inline math/binomial_calculate.cpp:40]: this needs to be documented as well

Comment (realstealthninja) [inline math/binomial_calculate.cpp:40]:
```suggestion
namespace math { 
namespace binomial {
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Pull request closed without merging**